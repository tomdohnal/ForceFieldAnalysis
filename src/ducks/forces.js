// @flow
import shortid from 'shortid';
import _ from 'lodash';

import { createReducer } from '../helpers';

export const ADD_FORCE: string = 'FORCES/ADD_FORCE';
export const DELETE_FORCE: string = 'FORCES/DELETE_FORCE';
export const SET_FORCE_NAME = 'FORCES/SET_FORCE_NAME';
export const INCREASE_FORCE = 'FORCES/INCREASE_FORCE';
export const DECREASE_FORCE = 'FORCES/DECREASE_FORCE';

export type Force = {
  id: string,
  driving: boolean,
  name: string,
  strength: number,
}

export type Forces = {
  [id: string]: Force,
};

type AddForceAction = {
  type: typeof ADD_FORCE,
  payload: Force,
}

type SetForceNameAction = {
  type: typeof SET_FORCE_NAME,
  payload: {
    id: string,
    name: string,
  }
}

type DeleteForceAction = {
  type: typeof DELETE_FORCE,
  payload: {
    id: string,
  }
}

type IncreaseForceAction = {
  type: typeof INCREASE_FORCE,
  payload: {
    id: string,
  }
}

type DecreaseForceAction = {
  type: typeof DECREASE_FORCE,
  payload: {
    id: string,
  }
}

const addForce = (name: string, strength: number, driving: boolean): AddForceAction => ({
  type: ADD_FORCE,
  payload: {
    id: shortid.generate(),
    driving,
    name,
    strength,
  },
});

export const addDrivingForce = (name: string, strength: number): AddForceAction => (
  addForce(name, strength, true)
);

export const addHinderingForce = (name: string, strength: number): AddForceAction => (
  addForce(name, strength, false)
);

export const deleteForce = (id: string): DeleteForceAction => ({
  type: DELETE_FORCE,
  payload: { id },
});

export const setForceName = (id: string, name: string): SetForceNameAction => ({
  type: SET_FORCE_NAME,
  payload: { id, name },
});

export const increaseForce = (id: string): IncreaseForceAction => ({
  type: INCREASE_FORCE,
  payload: { id },
});

export const decreaseForce = (id: string): DecreaseForceAction => ({
  type: DECREASE_FORCE,
  payload: { id },
});

export default createReducer({}, {
  [ADD_FORCE]: (state: Forces, action: DeleteForceAction) => ({
    ...state, [action.payload.id]: action.payload,
  }),
  [SET_FORCE_NAME]: (state: Forces, action: SetForceNameAction) => ({
    ...state,
    [action.payload.id]: { ...state[action.payload.id], name: action.payload.name },
  }),
  [DELETE_FORCE]: (state: Forces, action: DeleteForceAction) => (
    _.omit(state, action.payload.id)
  ),
  [INCREASE_FORCE]: (state: Forces, action: IncreaseForceAction) => ({
    ...state,
    [action.payload.id]:
      { ...state[action.payload.id], strength: state[action.payload.id].strength + 1 },
  }),
  [DECREASE_FORCE]: (state: Forces, action: DecreaseForceAction) => ({
    ...state,
    [action.payload.id]:
      { ...state[action.payload.id], strength: state[action.payload.id].strength - 1 },
  }),
});
