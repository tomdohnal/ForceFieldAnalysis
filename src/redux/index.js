// @flow
import { combineReducers, createStore } from 'redux';
import _ from 'lodash';

import { loadState, saveState } from './localStorage';

import forcesReducer, { type Forces } from '../ducks/forces';
import changeReducer, { type Change } from '../ducks/change';
import modeReducer, { type Mode } from '../ducks/mode';
import scaleReducer, { type Scale } from '../ducks/scale';

export type ReduxState = {
  forces: Forces,
  change: Change,
  mode: Mode,
  scale: Scale,
}

export type PersistedState = {
  forces: Forces,
  change: Change,
  mode: Mode,
}

export const RESET_STATE = 'RESET_STATE';

export type ResetStateAction = {
  type: typeof RESET_STATE
}

export const resetState = (): ResetStateAction => ({ type: RESET_STATE });

const appReducer = combineReducers({
  forces: forcesReducer,
  change: changeReducer,
  mode: modeReducer,
  scale: scaleReducer,
});

const rootReducer = (state, action) => {
  if (action.type === RESET_STATE) {
    return appReducer({}, action);
  }

  return appReducer(state, action);
};

const persistedState = loadState();

export const store = createStore(rootReducer, persistedState);

store.subscribe(_.throttle(() => {
  const { forces, change, mode } = store.getState();
  saveState({ forces, change, mode });
}, 1000));
