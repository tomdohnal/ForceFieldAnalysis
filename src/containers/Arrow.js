// @flow
import React, { Component } from 'react';
import { Form, Input, Button, Icon, Divider } from 'semantic-ui-react';
import EditIcon from 'react-icons/lib/ti/edit';
import { connect } from 'react-redux';

import { MAX_STRENGTH, MIN_STRENGTH } from '../constants/index';
import { COLORS } from '../constants/styles';
import { type Force, deleteForce, setForceName, increaseForce, decreaseForce } from '../ducks/forces';
import { Text, Box } from '../components/common/index';
import { Rectangle, Triangle } from '../components/svg/index';
import TextBubble from '../components/TextBubble';

type Props = {
  color: string,
  force: Force,
  deleteForce: (id: string) => void,
  setForceName: (id: string, name: string) => void,
  increaseForce: (id: string) => void,
  decreaseForce: (id: string) => void,
}

type State = {
  editArrow: boolean,
}

class Arrow extends Component<Props, State> {
  state = {
    editArrow: false,
  };

  onDeleteArrowIconClick = () => {
    this.props.deleteForce(this.props.force.id);
  };

  onIncreaseArrowIconClick = () => {
    if (this.props.force.strength !== MAX_STRENGTH) {
      this.props.increaseForce(this.props.force.id);
    }
  };

  onDecreaseArrowIconClick = () => {
    if (this.props.force.strength !== MIN_STRENGTH) {
      this.props.decreaseForce(this.props.force.id);
    }
  };

  onEditArrowInputChange = (event, data) => {
    this.props.setForceName(this.props.force.id, data.value);
  };

  onEditArrowNameClick = () => {
    this.setState({ editArrow: true });
  };

  onEditArrowNameFormSubmit = () => {
    this.setState({ editArrow: false });
  };

  render() {
    const { force: { driving, strength, name }, color } = this.props;

    const arrowUnitWidth = 50;
    const arrowWidth = strength * arrowUnitWidth;
    const arrowHeight = 80;
    const rectangleHeight = arrowHeight - 40;
    const triangleWidth = 36;

    return (
      <Box>
        <TextBubble color={color} left={!driving}>
          {this.state.editArrow ?
            <Form onSubmit={this.onEditArrowNameFormSubmit}>
              <Input autoFocus size="small" value={name} onChange={this.onEditArrowInputChange} action>
                <input style={{ width: 'inherit' }} />
                <Button size="small">Save changes</Button>
              </Input>
            </Form>
            :
            <Box>
              <Text fontSize="15px">{name}</Text>
              <EditIcon
                size="18"
                style={{ verticalAlign: 'top', marginLeft: '4px', cursor: 'pointer' }}
                onClick={this.onEditArrowNameClick}
              />
            </Box>
          }
        </TextBubble>
        <Box position="relative">
          <Icon
            name="delete"
            size="big"
            style={{
              position: 'absolute',
              top: 'calc(50% - 14px)',
              color: COLORS.RED,
              cursor: 'pointer',
              [driving ? 'right' : 'left']: `${arrowWidth}px`,
            }}
            onClick={this.onDeleteArrowIconClick}
          />
          <svg height={arrowHeight} width={arrowWidth} viewBox={`0 0 ${arrowWidth} ${arrowHeight}`}>
            <Rectangle
              width={arrowWidth - triangleWidth}
              height={rectangleHeight}
              color={color}
              x={!driving ? triangleWidth : 0}
              y={(arrowHeight - rectangleHeight) / 2}
            />
            <Triangle
              width={triangleWidth}
              height={arrowHeight}
              x={`${!driving ? triangleWidth : arrowWidth - triangleWidth},0`}
              y={`${!driving ? triangleWidth : arrowWidth - triangleWidth},${arrowHeight}`}
              z={`${!driving ? 0 : arrowWidth},${arrowHeight / 2}`}
              color={color}
            />
          </svg>
          <Box
            position="absolute"
            top={`${arrowHeight - ((arrowHeight - rectangleHeight) / 2)}px`}
            right={driving && `${triangleWidth}px`}
            left={!driving && `${triangleWidth}px`}
          >
            <Icon
              name="plus"
              size="large"
              disabled={strength === MAX_STRENGTH}
              style={{ cursor: `${strength < MAX_STRENGTH ? 'pointer' : 'initial'}` }}
              onClick={this.onIncreaseArrowIconClick}
            />
            <Text fontSize="18px">{strength}</Text>
            <Icon
              name="minus"
              size="large"
              disabled={strength === MIN_STRENGTH}
              style={{ cursor: `${strength > MIN_STRENGTH ? 'pointer' : 'initial'}` }}
              onClick={this.onDecreaseArrowIconClick}
            />
          </Box>
        </Box>
        <Divider hidden />
      </Box>
    );
  }
}

const actionCreators = { deleteForce, setForceName, increaseForce, decreaseForce };

export default connect(undefined, actionCreators)(Arrow);
