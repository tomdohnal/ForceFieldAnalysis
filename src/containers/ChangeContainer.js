// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Input, Button } from 'semantic-ui-react';

import { COLORS } from '../constants/styles';
import type { ReduxState } from '../redux/index';
import { setChangeName, setChangeDescription, type Change } from '../ducks/change';
import { Box, Text } from '../components/common/index';

type Props = {
  change: Change,
  onNameChange: (name: string) => void,
  onDescriptionChange: (description: string) => void,
};

type State = {
  editChangeName: boolean,
  editChangeDescription: boolean,
};

class ChangeContainer extends Component<Props, State> {
  state = {
    editChangeName: !this.props.change.name,
    editChangeDescription: !this.props.change.description,
  };

   onButtonClick = editChangeBoolean => (): void => {
     this.setState({ [editChangeBoolean]: false });
   };

   renderForm(onChange: (value: string) => void, onButtonClick: (value: string) => void) {
     return (
       <Box>
         <Input onChange={onChange} />
         <Button onClick={onButtonClick} />
       </Box>
     );
   }

   render() {
     const { change: { name, description }, onNameChange, onDescriptionChange } = this.props;

     return (
       <Box padding="20px 16px" color={COLORS.BLUE} height="100%">
         {this.state.editChangeName ?
           this.renderForm(onNameChange, this.onButtonClick('editChangeName')) :
           <Text>{name}</Text>
         }
         {this.state.editChangeDescription ?
           this.renderForm(onDescriptionChange, this.onButtonClick('editChangeDescription')) :
           <Text>{description}</Text>
         }
       </Box>
     );
   }
}

const mapStateToProps = (state: ReduxState) => ({
  change: state.change,
});

const mapDispatchToProps = dispatch => ({
  onNameChange: (event, data) => (dispatch(setChangeName(data.value))),
  onDescriptionChange: (event, data) => (dispatch(setChangeDescription(data.value))),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeContainer);
