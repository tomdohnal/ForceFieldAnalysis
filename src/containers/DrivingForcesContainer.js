// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Input, Button } from 'semantic-ui-react';

import DrivingForceArrow from '../components/DrivingForceArrow';
import { addDrivingForce, type Forces } from '../ducks/forces';

type State = {
  newDrivingForceName: string,
  newDrivingForceError: boolean,
  newDrivingForceErrorText: string,
};

type Props = {
  forces: Forces,
  addDrivingForce: (name: string, strength: number) => void,
};

class DrivingForcesContainer extends Component<Props, State> {
  state = {
    newDrivingForceName: '',
    newDrivingForceError: false,
    newDrivingForceErrorText: '',
  };

  onNewDrivingForceInputChange = (e) => {
    this.setState({
      newDrivingForceName: e.target.value,
      newDrivingForceError: false,
      newDrivingForceErrorText: '',
    });
  };

  onNewDrivingForceButtonClick = () => {
    if (!this.state.newDrivingForceName) {
      this.setState({
        newDrivingForceError: true,
        newDrivingForceErrorText: 'Enter the description for the force.',
      });
    } else {
      this.setState({
        newDrivingForceName: '',
      });

      this.props.addDrivingForce(this.state.newDrivingForceName, 2);
    }
  };

  renderForces() {
    return this.props.forces.map(force => (
      <DrivingForceArrow strength={force.strength} key={force.id} />));
  }

  render() {
    return (
      <div>
        {this.renderForces()}
        <Input
          onChange={this.onNewDrivingForceInputChange}
          value={this.state.newDrivingForceName}
          error={this.state.newDrivingForceError}
        />
        <Button onClick={this.onNewDrivingForceButtonClick}>New driving force</Button>
        <span>{this.state.newDrivingForceErrorText}</span>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  forces: state.forces.filter(force => force.driving === true),
});

export default connect(mapStateToProps, { addDrivingForce })(DrivingForcesContainer);
