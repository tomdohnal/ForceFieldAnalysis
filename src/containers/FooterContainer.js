// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Grid } from 'semantic-ui-react';

import { type Force } from '../ducks/forces';
import { type ReduxState } from '../redux';
import Footer from '../components/Footer';
import Text from '../components/Text';

type Props = {
  drivingForcesCount: number,
  hinderingForcesCount: number,
}

const FooterContainer = ({
  drivingForcesCount, hinderingForcesCount,
}: Props) => {
  let centerText: string = '';
  if (drivingForcesCount === 0 && hinderingForcesCount === 0) {
    centerText = 'Create some forces and start the analysis!';
  } else if (drivingForcesCount === hinderingForcesCount) {
    centerText = 'It\'s a tie!';
  } else if (drivingForcesCount > hinderingForcesCount) {
    centerText = 'Let\'s make the change happen!';
  } else {
    centerText = 'No way! Status quo wins...';
  }

  return (
    <Footer>
      <Grid>
        <Grid.Column width={6}>
          <Text fontSize="54px" color="#FFF">Driving forces: {drivingForcesCount}</Text>
        </Grid.Column>
        <Grid.Column width={4} textAlign="center">
          <Text fontSize="24px" color="#FFF">{centerText}</Text>
        </Grid.Column>
        <Grid.Column width={6} textAlign="right">
          <Text fontSize="54px" color="#FFF">Hindering forces: {hinderingForcesCount}</Text>
        </Grid.Column>
      </Grid>
    </Footer>
  );
};


const mapStateToProps = (state: ReduxState) => {
  const countForceTotal = (driving: boolean) => (total: number, force: Force) => {
    if (force.driving === driving) {
      return total + force.strength;
    }

    return total;
  };

  return {
    drivingForcesCount: state.forces.length ?
      state.forces.reduce(countForceTotal(true), 0) : 0,
    hinderingForcesCount: state.forces.length ?
      state.forces.reduce(countForceTotal(false), 0) : 0,
  };
};

export default connect(mapStateToProps)(FooterContainer);
