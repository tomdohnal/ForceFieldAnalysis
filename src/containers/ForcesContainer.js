// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Form, Input, Button, Select, Label } from 'semantic-ui-react';

import DrivingForceArrow from '../components/DrivingForceArrow';
import HinderingForceArrow from '../components/HinderingForceArrow';
import { Box } from '../components/common';
import { addDrivingForce, addHinderingForce, type Forces } from '../ducks/forces';
import { type ReduxState } from '../redux';

type State = {
  newForceName: string,
  newForceError: boolean,
  newForceErrorText: string,
  newForceStrength: number,
};

type Props = {
  driving: boolean,
  submitButtonText: string,
  forces: Forces,
  addForce: (name: string, strength: number) => void,
};

class ForcesContainer extends Component<Props, State> {
  defaultState = {
    newForceName: '',
    newForceError: false,
    newForceErrorText: '',
    newForceStrength: 1,
  };

  state = this.defaultState;

  onNewForceInputChange = (e, data) => {
    this.setState({
      newForceName: data.value,
      newForceError: false,
      newForceErrorText: '',
    });
  };

  onNewForceStrengthDropdownChange = (e, data) => {
    this.setState({
      newForceStrength: data.value,
    });
  };

  onNewForceButtonClick = () => {
    if (!this.state.newForceName) {
      this.setState({
        newForceError: true,
        newForceErrorText: 'Enter the description for the force.',
      });
    } else {
      this.props.addForce(this.state.newForceName, this.state.newForceStrength);

      this.setState(this.defaultState);
    }
  };

  getStrengthOptions() {
    let options = [];

    for (let i = 1; i <= 8; i += 1) {
      options = [...options, { key: i, text: i, value: i }];
    }

    return options;
  }

  renderForces() {
    return this.props.forces.map((force) => {
      if (this.props.driving) {
        return <DrivingForceArrow strength={force.strength} key={force.id} />;
      }
      return <HinderingForceArrow strength={force.strength} key={force.id} />;
    });
  }

  render() {
    return (
      <Box padding="0 16px">
        {this.renderForces()}
        <Form>
          <Form.Group inline>
            <Form.Field>
              <label>Force name:</label>
              {this.state.newForceError ? <Label basic color="red" pointing="below">{this.state.newForceErrorText}</Label> : ''}
              <Input
                onChange={this.onNewForceInputChange}
                value={this.state.newForceName}
                error={this.state.newForceError}
              />
            </Form.Field>
            <Form.Field>
              <label>Force strength</label>
              <Select
                compact
                labeled
                onChange={this.onNewForceStrengthDropdownChange}
                options={this.getStrengthOptions()}
                value={this.state.newForceStrength}
              />
            </Form.Field>
          </Form.Group>
          <Form.Field>
            <Button onClick={this.onNewForceButtonClick}>{this.props.submitButtonText}</Button>
          </Form.Field>
        </Form>
      </Box>
    );
  }
}

const mapStateToProps = (state: ReduxState, ownProps: Props) => ({
  forces: state.forces.filter(force => force.driving === ownProps.driving),
});

const mapDispatchToProps = (dispatch, ownProps: Props) => ({
  addForce: (name: string, strength: number) => (
    ownProps.driving ?
      dispatch(addDrivingForce(name, strength)) :
      dispatch(addHinderingForce(name, strength))
  ),
});

export default connect(mapStateToProps, mapDispatchToProps)(ForcesContainer);
