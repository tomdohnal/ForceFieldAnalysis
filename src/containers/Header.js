// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Grid } from 'semantic-ui-react';

import { Box, Text } from '../components/common/index';
import { COLORS, HEADER_HEIGHT } from '../constants/styles';
import AboutBox from '../components/AboutBox';
import { resetState, type ResetStateAction, type ReduxState } from '../redux';
import { switchToCompleteMode, switchToEditMode, type SwitchToCompleteModeAction, type SwitchToEditModeAction, type Mode, EDIT_MODE } from '../ducks/mode';

type Props = {
  resetState: () => ResetStateAction,
  switchToCompleteMode: () => SwitchToCompleteModeAction,
  switchToEditMode: () => SwitchToEditModeAction,
  appMode: Mode,
}

type State = {
  showAbout: boolean,
}

export class Header extends Component<Props, State> {
  state = {
    showAbout: false,
  };

  onResetButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.resetState();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onMakeDecisionButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.switchToCompleteMode();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onEditForcesButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.switchToEditMode();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onAboutLinkClick = () => {
    this.setState({ showAbout: true });
  };

  onCloseAboutLinkClick = () => {
    this.setState({ showAbout: false });
  };

  render() {
    return (
      <Box height={HEADER_HEIGHT} paddingTop="16px">
        <Grid columns={3}>
          <Grid.Row>
            <Grid.Column>
              <Box paddingLeft="16px">
                <Button
                  onClick={this.onResetButtonClick}
                  style={{ color: COLORS.WHITE, background: COLORS.RED }}
                >
                  Reset analysis
                </Button>
                {this.props.appMode === EDIT_MODE ?
                  <Button onClick={this.onMakeDecisionButtonClick}>Make decision</Button>
                  :
                  <Button onClick={this.onEditForcesButtonClick}>Edit forces</Button>
                }
              </Box>
            </Grid.Column>
            <Grid.Column>
              <Box textAlign="center">
                <Text fontSize="24px">Force Field Analysis Tool</Text>
              </Box>
            </Grid.Column>
            <Grid.Column floated="right">
              {this.state.showAbout ?
                <AboutBox onCloseLinkClick={this.onCloseAboutLinkClick} />
                :
                <Text
                  onClick={this.onAboutLinkClick}
                  position="absolute"
                  right="30px"
                  clickable
                >About
                </Text>
              }
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Box>
    );
  }
}

export const mapStateToProps = (state: ReduxState) => ({ appMode: state.mode });

export default connect(
  mapStateToProps, { resetState, switchToCompleteMode, switchToEditMode },
)(Header);
