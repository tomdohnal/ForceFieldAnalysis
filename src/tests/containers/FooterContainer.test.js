import React from 'react';
import { shallow } from 'enzyme';

import { FooterContainer, mapStateToProps } from '../../containers/FooterContainer';
import { store, RESET_STATE } from '../../redux';
import { addDrivingForce, addHinderingForce } from '../../ducks/forces';

describe('FooterContainer', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <FooterContainer
        drivingForcesCount={0}
        hinderingForcesCount={0}
      />,
    ).dive();

    expect(tree).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    beforeEach(() => {
      store.dispatch({ type: RESET_STATE });
    });

    it('it counts driving forces correctly', () => {
      store.dispatch(addDrivingForce('force name', 5));
      store.dispatch(addDrivingForce('force name', 5));
      store.dispatch(addHinderingForce('force name', 5));
      store.dispatch(addHinderingForce('force name', 5));

      const { drivingForcesCount } = mapStateToProps(store.getState());

      expect(drivingForcesCount).toBe(10);
    });

    it('it counts hindering forces correctly', () => {
      store.dispatch(addHinderingForce('force name', 5));
      store.dispatch(addHinderingForce('force name', 5));
      store.dispatch(addDrivingForce('force name', 5));
      store.dispatch(addDrivingForce('force name', 5));

      const { hinderingForcesCount } = mapStateToProps(store.getState());

      expect(hinderingForcesCount).toBe(10);
    });
  });
});
