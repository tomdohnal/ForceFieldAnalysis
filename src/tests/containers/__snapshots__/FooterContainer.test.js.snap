// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterContainer renders correctly 1`] = `
<footerStyles>
  <Grid
    columns={3}
  >
    <GridColumn>
      <text
        color="#FFF"
        fontSize="54px"
      >
        Driving forces: 
        0
      </text>
    </GridColumn>
    <GridColumn
      textAlign="center"
    >
      <text
        color="#FFF"
        fontSize="24px"
      >
        Create forces to start the analysis!
      </text>
    </GridColumn>
    <GridColumn
      textAlign="right"
    >
      <text
        color="#FFF"
        fontSize="54px"
      >
        Hindering forces: 
        0
      </text>
    </GridColumn>
  </Grid>
</footerStyles>
`;
